src/historywork/BatchWork.cpp

```diff
@@ -0,0 +1,127 @@
+// Copyright 20158 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "BatchWork.h"
+#include "catchup/CatchupManager.h"
+#include "lib/util/format.h"
+#include "main/Application.h"
+#include "util/Logging.h"
+#include <history/FileTransferInfo.h>
+
+namespace stellar
+{
+
+BatchWork::BatchWork(Application& app, WorkParent& parent,
```

Comment:
`BatchWork` should introduce the building block for batching - this one does too much as it hardcodes "download" (so you cannot batch generic work with it).